### ISSUE TITLE
Add rationale to some guidelines. 

### ISSUE THREAD

**Issue by Koded** [WhiteHouse/api-standards#40]
I have found this a great resource when defining REST APIs.  One small thing.

In the [responses](https://github.com/WhiteHouse/api-standards#responses) section, it is stated that "No values in keys".  Whilst I agree with this, and get some way in being able to clearly articulate why, it would be good to ensure that the rationale behind the standard is captured and documented.

**Comment by adam-lynch** [comment]
I agree. Why is this a good practice by the way? What if it was an object with each sub-object still containing an `id` property as well?

**Comment by smrgeoinfo** [comment]
Can you provide an example of some JSON for the situation you suggest?    In general, I think its a good idea not to put values in keys under the interpretation that they keys are supposed to provide some indication of what the value means.

**Comment by adam-lynch** [comment]
@smrgeoinfo It's easier to get something by ID, etc. then. See https://github.com/json-api/json-api/issues/1170 for an example.
